projects/storefrontlib/src/cms-components/checkout/components/checkout-progress/checkout-progress-mobile-bottom/checkout-progress-mobile-bottom.component.ts

```diff
@@ -15,35 +9,17 @@ import { CheckoutStepService } from '../../../services/checkout-step.service';
   templateUrl: './checkout-progress-mobile-bottom.component.html',
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class CheckoutProgressMobileBottomComponent
-  implements OnInit, OnDestroy {
-  constructor(
-    protected checkoutStepService: CheckoutStepService,
-    protected cdr: ChangeDetectorRef
-  ) {}
+export class CheckoutProgressMobileBottomComponent {
+  constructor(protected checkoutStepService: CheckoutStepService) {}
 
-  steps: CheckoutStep[];
+  steps$: Observable<
+    CheckoutStep[]
+  > = this.checkoutStepService.steps$.asObservable();
```

Comment:
let's move this to the service and call the a function that returns exactly above

```
  getSteps(): Observable<CheckoutStep[]> {
    return this.steps$.asObservable();
  }
```

so that customers won't have to do it that way when using the service